bin/fetch-proxy

```diff
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/bin/sh
 
 # If the first argument to this script is "latest" or unset, it fetches the
 # latest proxy binary via build.l5d.io/linkerd2-proxy/latest.txt. If it's set to
```

Comment:
While you're updating this file, it looks like this comment is actually out of date.  Would you mind updating it while you're here?  This script no longer users `build.l5d.io` but instead fetches the latest proxy release from the linkerd2-proxy github releases.

Reply:
No problem, I'll see to that.

Reply:
Done.